src/CorrelationId/CorrelationIdMiddleware.cs

```diff
@@ -35,7 +35,7 @@ public CorrelationIdMiddleware(RequestDelegate next, IOptions<CorrelationIdOptio
                 context.TraceIdentifier = correlationId;
             }
 
-            correlationContextFactory.Create(context.TraceIdentifier);
+            correlationContextFactory.Create(context.TraceIdentifier, _options.Value.Header);
```

Comment:
Why not just pass the entire `context` and `_options.Value` here? So you don't have to rev the interface every time someone needs a new bit of information in the factory? 😄

Reply:
BTW, this is a breaking change (why I'm here) and would've probably warranted a major bump 😝

Reply:
Fair point. From the HttpContext I didn't foresee anything else being grabbed so I was fairly happy taking just the TraceIdentifier. That said... you never know 😄. For the options it makes more sense if it's likely that there are more settings that will end up exposed on the CorrelationContext. Again, at this point I'm not imaging any.